src/irmin-pack/irmin_pack.ml

```diff
@@ -399,6 +399,52 @@ struct
     end
   end
 
+  let integrity_check (t : X.Repo.t) =
+    Fmt.pr "running the integrity check\n%!";
+    let commits = ref 0 in
+    let contents = ref 0 in
+    let nodes = ref 0 in
+    let pp_stats ppf () =
+      Fmt.pf ppf "%4dk blobs / %4dk trees / %4dk commits" (!contents / 1000)
+        (!nodes / 1000) (!commits / 1000)
+    in
+    let pr_stats () = Fmt.epr "\r%a%!" pp_stats () in
+    let count_increment count =
+      incr count;
+      if !count mod 100 = 0 then pr_stats ()
+    in
+    Index.iter
+      (fun k (offset, length, m) ->
+        match m with
+        | 'B' ->
+            Lwt.async (fun () ->
+                let () =
+                  X.Repo.contents_t t
+                  |> X.Contents.CA.integrity_check ~offset ~length k
+                in
+                count_increment contents;
+                Lwt.return_unit )
+        | 'N' | 'I' ->
+            Lwt.async (fun () ->
+                let () =
+                  snd @@ X.Repo.node_t t
+                  |> X.Node.CA.integrity_check ~offset ~length k
+                in
+                count_increment nodes;
+                Lwt.return_unit )
+        | 'C' ->
+            Lwt.async (fun () ->
+                let () =
+                  snd @@ X.Repo.commit_t t
+                  |> X.Commit.CA.integrity_check ~offset ~length k
+                in
+                count_increment commits;
+                Lwt.return_unit )
+        | _ -> invalid_arg "unknown content type" )
+      t.index;
+    Unix.sleep 10;
```

Comment:
This looks odd

Reply:
Yes, good catch. I was convinced I'd gotten rid of this.